code/modules/antagonists/revolution/revolution.dm

```diff
@@ -116,54 +116,53 @@
 /datum/antagonist/rev/head/get_admin_commands()
 	. = ..()
 	. -= "Promote"
-	.["Take flash"] = CALLBACK(src,PROC_REF(admin_take_flash))
-	.["Give flash"] = CALLBACK(src,PROC_REF(admin_give_flash))
-	.["Repair flash"] = CALLBACK(src,PROC_REF(admin_repair_flash))
 	.["Demote"] = CALLBACK(src,PROC_REF(admin_demote))
 
-/datum/antagonist/rev/head/proc/admin_take_flash(mob/admin)
-	var/list/L = owner.current.get_contents()
-	var/obj/item/assembly/flash/flash = locate() in L
-	if (!flash)
-		to_chat(admin, "<span class='danger'>Deleting flash failed!</span>")
-		return
-	qdel(flash)
-
-/datum/antagonist/rev/head/proc/admin_give_flash(mob/admin)
-	//This is probably overkill but making these impact state annoys me
-	var/old_give_flash = give_flash
-	var/old_give_hud = give_hud
-	var/old_remove_clumsy = remove_clumsy
-	give_flash = TRUE
-	give_hud = FALSE
-	remove_clumsy = FALSE
-	equip_rev()
-	give_flash = old_give_flash
-	give_hud = old_give_hud
-	remove_clumsy = old_remove_clumsy
-
-/datum/antagonist/rev/head/proc/admin_repair_flash(mob/admin)
-	var/list/L = owner.current.get_contents()
-	var/obj/item/assembly/flash/flash = locate() in L
-	if (!flash)
-		to_chat(admin, "<span class='danger'>Repairing flash failed!</span>")
-	else
-		flash.burnt_out = FALSE
-		flash.update_icon()
-
 /datum/antagonist/rev/head/proc/admin_demote(datum/mind/target,mob/user)
 	message_admins("[key_name_admin(user)] has demoted [key_name_admin(owner)] from head revolutionary.")
 	log_admin("[key_name(user)] has demoted [key_name(owner)] from head revolutionary.")
 	demote()
 
+/obj/effect/proc_holder/spell/targeted/revolution_convert
+	name = "Revolutionary Conversion"
+	desc = "Converts a target into a revolutionary."
+	charge_max = 0
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "mindread"
+	action_background_icon_state = "bg_demon"
+	range = 3
+	clothes_req = FALSE
+
+
 /datum/antagonist/rev/head
 	name = "Head Revolutionary"
 	hud_type = "rev_head"
 	banning_key = ROLE_REV_HEAD
 	required_living_playtime = 4
 	var/remove_clumsy = FALSE
-	var/give_flash = FALSE
 	var/give_hud = TRUE
+	var/give_computer = TRUE
+
+
+/obj/effect/proc_holder/spell/targeted/revolution_convert/cast(list/targets, mob/user)
+	var/datum/antagonist/rev/rev_datum = user.mind.has_antag_datum(/datum/antagonist/rev)
+	if(targets.len > 1)
+		to_chat(user, "<span class='warning'>You can only convert one person at a time!</span>")
+		return
+	if(!rev_datum.can_be_converted(targets[1]))
+		to_chat(user, "<span class='warning'>You can't convert that person!</span>")
+		return
+	for(var/mob/living/carbon/target in targets)
+		for(var/i = 1 to 3)
+			user.say("I hereby convert you to the revolution!", forced = "revolution convertion")
```

Comment:
Also won't this just say "I hereby convert you to the revolution!" 3 times

Reply:
Yes it does, my idea is to create a string list and get a random thing!

Reply:
That sounds good, its what I would have suggested